pages/unitCard.tsx

```diff
@@ -1,40 +1,98 @@
 import { NextPage } from "next";
-import { getWeaponData, WeaponData } from "../src/unitStats/WeaponLib";
-import { DpsChart } from "../components/unitStats/DpsChart";
+import { DpsChart } from "../components/unitStats/dpsChart";
+import { EbpsType, getEbpsStats, setEbpsStats } from "../src/unitStats/mappingEbps";
+import { getSbpsStats, SbpsType, setSbpsStats } from "../src/unitStats/mappingSbps";
+import {
+  getWeaponStats,
+  setWeaponStats,
+  WeaponStats,
+  WeaponType,
+} from "../src/unitStats/mappingWeapon";
+import {
+  getUpgradesStats,
+  setUpgradesStats,
+  UpgradesType,
+} from "../src/unitStats/mappingUpgrades";
+
+type Locstring = {
+  id: number;
+  value: string;
+};
+
+let unitStatsLocString: Locstring[];
+
+export const resolveLocstring = (locstring: any) => {
+  const numId = parseInt(locstring?.locstring?.value);
+
+  let result = unitStatsLocString?.find((entry) => entry.id == numId)?.value;
+  if (!result) result = "No text found.";
+
+  return result;
+};
```

Comment:
Why not use the plain locstring JSON as it is instead of turning it into an array?

```ts
// Locstring example.
// {
//    "1234": "Value A",
//    "5678": "Value B"
// }

type LocstringSchema = {
  id: number;
  value: string;
};

type LocstringObjectSchema = {
  { locstring: LocstringSchema }
};

// The input comes from the weapon / ebps / sbps / upgrade json.
export const resolveLocstring = (inLocstring: LocstringObjectSchema) => {

  // unitStatsLocString is a object (Record<string, string>
  return unitStatsLocString[inLocstring?.locstring?.value] ?? "No text found.";
};
```

So we don't have to do the extra `find` in the array. Just use object indexing with the `?.` operator.

Reply:
Looks better. I'm gonna change that.